lib/ui/home/buttons/rectangular_button.dart

```diff
@@ -15,56 +16,62 @@ class RectangularButton extends StatefulWidget {
 }
 
 class _RectangularButtonState extends State<RectangularButton> {
-  late final Icon? icon = widget.imageKey != null ? Icon(
-    IconsHelper.getIcon(widget.imageKey!),
-    color: Colors.white,
-    size: 45,
-  ) : null;
-
-  late Color currentColor;
-  final borderRadius = BorderRadius.circular(8.0);
+  late Color buttonColor;
 
   @override
   void initState() {
     super.initState();
-    currentColor = widget.color;
+    buttonColor = widget.color;
   }
 
   void onTap() {
     setState(() {
-      // Here you are changing the color to grey when clicked
-      currentColor = Colors.grey; 
+      buttonColor = Colors.grey; // Change color to grey when clicked
     });
+    
     log("Clicked on ${widget.description}");
+
+    // Delay for a short time to visually see the color change
+    Future.delayed(Duration(milliseconds: 500), () {
```

Comment:
Adding delays on triggers is generally a bad practice causing unwanted behavior (when button remains pressed for example)